plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,299 @@
+	);
+
+	// List of assets to check.
+	$assets = array(
+		includes_url( 'js/wp-embed.min.js' ),
+		includes_url( 'css/buttons.min.css' ),
+		includes_url( 'fonts/dashicons.woff2' ),
+		includes_url( 'images/media/video.png' ),
+	);
+
+	/**
+	 * Filters the list of assets to check for far-future headers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string[] $assets List of asset URLs to check.
+	 */
+	$assets = apply_filters( 'perflab_ffh_assets_to_check', $assets );
+
+	// Check if far-future headers are enabled for all assets.
+	$results = perflab_ffh_check_assets( $assets );
+
+	if ( 'good' !== $results['final_status'] ) {
+		$result['status'] = $results['final_status'];
+		$result['label']  = __( 'Your site does not serve static assets with recommended far-future expiration headers', 'performance-lab' );
+
+		if ( count( $results['details'] ) > 0 ) {
+			$result['actions'] = sprintf(
+				'<p>%s</p>%s<p>%s</p>',
+				esc_html__( 'The following file types do not have the recommended far-future headers. Consider adding or adjusting Cache-Control or Expires headers for these asset types.', 'performance-lab' ),
+				perflab_ffh_get_extensions_table( $results['details'] ),
+				esc_html__( 'Note: "Conditionally cached" means that the browser can re-validate the resource using ETag or Last-Modified headers. This results in fewer full downloads but still requires the browser to make requests, unlike far-future expiration headers that allow the browser to fully rely on its local cache for a longer duration.', 'performance-lab' )
+			);
+		} else {
+			$result['actions'] = sprintf(
+				'<p>%s</p>',
+				esc_html__( 'Far-future Cache-Control or Expires headers can be added or adjusted with a small configuration change by your hosting provider.', 'performance-lab' )
+			);
+		}
+	}
+
+	return $result;
+}
+
+/**
+ * Checks if far-future expiration headers are enabled for a list of assets.
+ *
+ * @since n.e.x.t
+ *
+ * @param  string[] $assets List of asset URLs to check.
+ * @return array{final_status: string, details: array{filename: string, reason: string}[]} Final status and details.
+ */
+function perflab_ffh_check_assets( array $assets ): array {
+	$final_status = 'good';
+	$fail_details = array(); // Array of arrays with 'filename' and 'reason'.
+
+	foreach ( $assets as $asset ) {
+		$response = wp_remote_get( $asset, array( 'sslverify' => false ) );
+
+		// Extract filename from the URL.
+		$path_info = pathinfo( (string) wp_parse_url( $asset, PHP_URL_PATH ) );
+		$filename  = isset( $path_info['basename'] ) ? $path_info['basename'] : basename( $asset );
+
+		if ( is_wp_error( $response ) ) {
+			// Can't determine headers if request failed, consider it a fail.
+			$final_status   = 'recommended';
+			$fail_details[] = array(
+				'filename' => $filename,
+				'reason'   => __( 'Could not retrieve headers', 'performance-lab' ),
+			);
+			continue;
+		}
+
+		$headers = wp_remote_retrieve_headers( $response );
+		if ( ! is_object( $headers ) ) {
```

Comment:
Since `wp_remote_retrieve_headers()` can return an `array`:

```suggestion
		if ( count( $headers ) === 0 ) {
```

Reply:
>Fatal error: Uncaught Error: count(): Argument # 1 ($value) must be of type Countable|array, WpOrg\Requests\Utility\CaseInsensitiveDictionary given

This error is encountered when using `count`, below condition seems to solve this error.
`if  ( ! is_object( $headers ) && 0 === count( $headers ) )`

Reply:
It's surprising that `CaseInsensitiveDictionary` is not implemented as a `Countable`.